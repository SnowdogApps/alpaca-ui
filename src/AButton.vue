<script lang="ts">
export const variants = [
  'primary',
  'secondary',
  'tertiary',
  'transparent',
] as const
</script>

<script setup lang="ts">
type ButtonType = 'button' | 'submit' | 'reset'

type ButtonProps = {
  variant: typeof variants[number]
  tag: 'button' | 'a'
  type?: ButtonType
  disabled?: boolean
}

withDefaults(defineProps<ButtonProps>(), {
  variant: 'primary',
  tag: 'button',
  type: 'button',
  disabled: undefined,
})
</script>

<template>
  <component
    :is="tag"
    :type="tag === 'button' ? type : undefined"
    :disabled="disabled"
    class="inline-flex h-12 items-center justify-center px-6 font-bold uppercase transition-colors duration-200 ease-in focus:outline-focus"
    :class="[
      variant === 'primary' &&
        'bg-coral-300 hover:bg-coral-400 hover:text-white',
      variant === 'secondary' &&
        'border border-dark hover:bg-dark hover:text-white',
      variant === 'tertiary' && 'border border-coral-300 hover:bg-coral-300',
      variant === 'transparent' && 'hover:bg-gray-100',
      disabled &&
        'cursor-not-allowed border-0 bg-gray-100 hover:bg-gray-100 hover:text-gray-500',
    ]"
  >
    <slot />
  </component>
</template>
