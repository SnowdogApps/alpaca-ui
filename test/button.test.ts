import { mount } from '@vue/test-utils'
import AButton from '../src/AButton.vue'

test('mount component', async () => {
  expect(AButton).toBeTruthy()

  const wrapper = mount(AButton, {
    props: {
      color: 'red',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})
