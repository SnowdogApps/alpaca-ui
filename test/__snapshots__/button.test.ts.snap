// Vitest Snapshot v1

exports[`mount component 1`] = `"<button type=\\"button\\" class=\\"inline-flex h-14 items-center justify-center px-6 font-bold uppercase transition-colors duration-200 ease-in focus:outline-focus bg-coral-300 hover:bg-coral-400 hover:text-white\\" color=\\"red\\"></button>"`;
